### DISCUSSION TITLE
Does this library render the modal ALWAYS at the highest z-index?

### DISCUSSION THREAD

**Discussion by cwik-tech** [reactjs/react-modal#1032] [Q&A]
I am looking for a solution that will enable me passing the context of the modal into it from any component in the app and it would still be rendered at the highest z-index. Does this solution work like that? all the examples I have seen show local triggering of the modal so local z-index/div order will apply. If so, could you provide a link to example or docs? thx.

**Comment by sergei-maertens** [comment]
z-index overlay order is subject to the [stacking context](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_positioned_layout/Understanding_z-index/Stacking_context).

Combining this knowledge with the parent node prop(s) (https://reactcommunity.org/react-modal/#custom-parent), you can ensure that the modal gets rendered in a sufficiently high stacking context and will always be on top.

The simplest case is having the parent be appended to the document body, where it has a stacking context and positioned over its siblings (the preceding body children). Whatever stacking context then happens in a child is irrelevant, because the modal parent is positioned above all that.
